### ISSUE TITLE
bucket notification API: prefix and suffix don't support asterisk character

### ISSUE THREAD

**Issue by vadmeste** [minio/minio#2483]
```
$ mc listen myminio/testbucket/ --suffix "*.go"
mc: <ERROR> Cannot watch on the specified bucket. prefix rule value cannot exceed 1024 characters
```

**Comment by harshavardhana** [comment]
> $ mc listen myminio/testbucket/ --suffix "*.go"
> mc: <ERROR> Cannot watch on the specified bucket. prefix rule value cannot exceed 1024 characters

Does 's3' support?

**Comment by vadmeste** [comment]
> Does 's3' support?

yes, S3 supports asterisk character, by the way, it doesn't interpret it

**Comment by harshavardhana** [comment]
> yes, S3 supports asterisk character, by the way, it doesn't interpret it

Suffix doesn't have regex support, also the error message here is wrong.

**Comment by harshavardhana** [comment]
Fixed by #2487
